apps/server/src/modules/board/repo/board-do.repo.ts

```diff
@@ -0,0 +1,86 @@
+import { Utils } from '@mikro-orm/core';
+import { EntityManager } from '@mikro-orm/mongodb';
+import { Injectable } from '@nestjs/common';
+import { AnyBoardDo, BoardNode, EntityId } from '@shared/domain';
+import { BoardDoBuilder } from '@shared/domain/entity/boardnode/board-do.builder';
+import { BoardNodeBuilderImpl } from '@shared/domain/entity/boardnode/board-node-builder-impl';
+import { BoardNodeRepo } from './board-node.repo';
+
+@Injectable()
+export class BoardDoRepo {
+	constructor(private readonly em: EntityManager, private readonly boardNodeRepo: BoardNodeRepo) {}
+
+	async findById(id: EntityId): Promise<AnyBoardDo> {
+		const boardNode = await this.em.findOneOrFail(BoardNode, id);
+		const descendants = await this.boardNodeRepo.findDescendants(boardNode);
+		const domainObject = new BoardDoBuilder(descendants).buildDomainObject(boardNode);
+
+		return domainObject;
+	}
+
+	async findByIds(ids: EntityId[]): Promise<AnyBoardDo[]> {
+		const boardNodes = await this.em.find(BoardNode, { id: { $in: ids } });
+
+		const childrenMap = await this.boardNodeRepo.findChildrenOfMany(boardNodes);
+
+		const domainObjects = boardNodes.map((boardNode) => {
+			const children = childrenMap[boardNode.pathOfChildren];
+			const domainObject = new BoardDoBuilder(children).buildDomainObject(boardNode);
+			return domainObject;
+		});
+
+		return domainObjects;
+	}
+
+	async findParentOfId(childId: EntityId): Promise<AnyBoardDo> {
```

Comment:
This is also very generic. I think we should be more specific in terms of domain objects.

Reply:
I dont understand this point?

just like we can get a node by id, we also need to be able to get the parent of a node. We dont know what type the parent might have in the future, and it works the same for any type of node so there is no point in making it specific to a type.